feat!: Refactor Testimonial and Avatar to Emotion

rogervera opened the pull request:
### Overview
<!--- CHANGELOG-DESCRIPTION -->
Refactors Testimonial and Avatar to Emotion.
<!--- END-CHANGELOG-DESCRIPTION -->
Historically, the only `medium` size use case was for the pilot that only appears on the data science career path marketing page. This component was frequently used alongside `useBreakpoint` hooks, to render a small or large size with a short and/or long quote.

Now, the testimonial uses a Flex layout to be responsive within it's container. 
**Also**, an optional `largeQuote` prop handles changing the text based on breakpoints.

### PR Checklist

- [x] Related to designs: [Figma](https://www.figma.com/file/dVmRmgyC7z5YcjBgiVzm5I/Learners-Like-You-(Master-%40-6ea7a37)?node-id=2%3A19208)
- [x] Related to JIRA ticket: [REACH-119](https://codecademy.atlassian.net/browse/REACH-119), [473](https://codecademy.atlassian.net/browse/REACH-473), [474](https://codecademy.atlassian.net/browse/REACH-474), and [478](https://codecademy.atlassian.net/browse/REACH-478)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [ ] Desk check'd by Tímea (original designer)


### Before
<img width="976" alt="Screen Shot 2021-04-12 at 8 51 49 AM" src="https://user-images.githubusercontent.com/17210163/114397129-5765e180-9b6c-11eb-88e8-bc29918e81f5.png">

### After
<img width="943" alt="Screen Shot 2021-04-12 at 8 50 53 AM" src="https://user-images.githubusercontent.com/17210163/114396994-369d8c00-9b6c-11eb-90eb-c732b510886b.png">


<!--
Merging your changes

1. Follow the [PR Title Guide](https://github.com/Codecademy/client-modules#pr-title-guide), the title (which becomes the commit message) determines the version bump for the packages you changed.

2. Wrap the text describing your change in more detail in the "CHANGELOG-DESCRIPTION" comment tags above, this is what will show up in the changelog!

3. DO NOT MERGE MANUALLY! When you are ready to merge and publish your changes, add the "Ship It" label to your Pull Request. This will trigger the merge process as long as all checks have completed, if the checks haven't completed the branch will be merged when they all pass.

**IMPORTANT:** If your PR contains breaking changes, please remember to follow the instructions for breaking changes!
-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://604e37ff64e222335b8ef8a9--gamut-preview.netlify.app>https://604e37ff64e222335b8ef8a9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/604e37ff64e222335b8ef8a9>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.16-alpha.dbb9be.0<br/>@codecademy/gamut-labs@11.2.4-alpha.dbb9be.0<br/>@codecademy/styleguide@30.6.1-alpha.dbb9be.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://604e4edae8131884f74bbe4d--gamut-preview.netlify.app>https://604e4edae8131884f74bbe4d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/604e4edae8131884f74bbe4d>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.16-alpha.2cea65.0<br/>@codecademy/gamut-labs@11.2.4-alpha.2cea65.0<br/>@codecademy/styleguide@30.6.1-alpha.2cea65.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://604f4ceaa9112ec4384fa092--gamut-preview.netlify.app>https://604f4ceaa9112ec4384fa092--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/604f4ceaa9112ec4384fa092>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/Testimonial/index.tsx:
```diff
@@ -1,75 +1,196 @@
+    grid-auto-flow: column;
+    gap: ${spacing[16]};
+
+    ${mediaQueries.sm} {
```
There's probably a way to do something like `${!compact && ` so we don't end up with a bunch of `grid-auto-flow: false;`s. 

Looking through https://gamut.codecademy.com/storybook/?path=/docs/foundations-layout--page, how about `atViewport`?

JoshuaKGoldberg reviewed (changes requested):
LGTM in general and the story is very nice 😄 . I think this at risk of producing invalid CSS, so can we try the `atViewport` approach or similar?

rogervera commented:
> LGTM in general and the story is very nice 😄 . I think this at risk of producing invalid CSS, so can we try the `atViewport` approach or similar?

Just met with Aaron to go over the finer details on improvements here. Stayed tuned!

aresnik11 commented on packages/gamut-labs/src/brand/Avatar/index.tsx:
```diff
@@ -1,46 +1,76 @@
+
+const Image = styled.img();
+
+const AvatarContainer = styled.div(({ mode }: { mode: VisualTheme }) => {
+  const { shadowColor } = modes[mode];
```
TIL!

aresnik11 commented on packages/gamut-labs/src/brand/Testimonial/index.tsx:
```diff
@@ -1,75 +1,198 @@
-export type Testimonial = {
+const modes = {
+  light: {
+    cardBackground: brandColors.darkBlue, // recolor
+    textColor: colors.white,
```
curious why we're using both `brandColors` and `colors` here? shouldn't the `brandColors` be encompassed in `colors`?

rogervera replied:
That's a good nit. White is the same in both, so I wanted to atleast move the needle a tiny bit.

aresnik11 commented on packages/gamut-labs/src/brand/Testimonial/index.tsx:
```diff
@@ -1,75 +1,198 @@
+  width: calc(100% - 280px);
+`;
+
+export const ShortQuote = styled.span`
+  display: initial;
+  ${mediaQueries.md} {
+    display: none;
+  }
+`;
+
+export const LongQuote = styled.span`
+  display: none;
+  ${mediaQueries.md} {
+    display: initial;
+  }
+`;
```
wondering if instead of this we can do something along the lines of
```
<Box as="span" display={{ base: 'none', md: 'initial' }}>
```
i've seen @codecaaron do this but not sure if it is best practice

aresnik11 replied:
oh you are below! so i think `LongQuote` and `ShortQuote` here aren't used and can be removed

rogervera replied:
Totally right. Will remove.

aresnik11 commented on packages/gamut-labs/src/brand/Testimonial/index.tsx:
```diff
@@ -1,75 +1,198 @@
+  `;
+});
+
+const TestimonialCard = styled(GridBox)();
```
is this necessary? i'm guessing it's to be more explicit with the naming but not sure if that is really needed

rogervera replied:
Using this as an explicit selector in the parent TestimonialWrapper
```jsx
${TestimonialCard} {
   align-content: center;
```

aresnik11 commented on packages/gamut-labs/src/brand/Quote/__tests__/Quote-test.tsx:
```diff
@@ -5,7 +5,7 @@ import { Quote } from '..';
 describe('Quote', () => {
   it('should render text when it is provided', () => {
-    const wrapper = render(<Quote text="Timshel!" />);
+    const wrapper = render(<Quote>Timshel!</Quote>);
```
according to the build errors, we need to pass in the `mode` prop here

aresnik11 replied:
```
src/brand/Quote/__tests__/Quote-test.tsx(8,29): error TS2741: Property 'mode' is missing in type '{ children: string; }' but required in type 'QuoteProps'.
```

aresnik11 commented on packages/gamut-labs/src/brand/Testimonial/__tests__/Testimonial-test.tsx:
```diff
@@ -13,87 +12,24 @@ const exampleTestmonial = {
   it('does _not_ render the Avatar component when an imageUrl prop is _not_ present', () => {
     const wrapper = mount(
-      <Testimonial testimonial={exampleTestmonial} size="small" theme="dark" />
+      <Testimonial {...exampleTestmonial} size="small" theme="dark" />
```
anddd `size` needs to be removed here and in the test above
https://app.circleci.com/pipelines/github/Codecademy/client-modules/11450/workflows/6836b604-1fc6-4897-a49e-73d07070c7c3/jobs/74341
```
src/brand/Testimonial/__tests__/Testimonial-test.tsx(20,9): error TS2322: Type '{ imageUrl: string; size: string; theme: string; firstName: string; lastName: string; occupation: string; quote: string; }' is not assignable to type 'IntrinsicAttributes & TestimonialProps & { children?: ReactNode; }'.
  Property 'size' does not exist on type 'IntrinsicAttributes & TestimonialProps & { children?: ReactNode; }'.
src/brand/Testimonial/__tests__/Testimonial-test.tsx(30,43): error TS2322: Type '{ size: string; theme: string; firstName: string; lastName: string; occupation: string; quote: string; }' is not assignable to type 'IntrinsicAttributes & TestimonialProps & { children?: ReactNode; }'.
  Property 'size' does not exist on type 'IntrinsicAttributes & TestimonialProps & { children?: ReactNode; }'.
```

aresnik11 commented on packages/gamut-labs/src/brand/Avatar/index.tsx:
```diff
@@ -1,46 +1,76 @@
 
-import styles from './styles.module.scss';
+const modes = {
+  dark: {
+    shadowColor: brandColors.orange,
+  },
+  light: {
+    shadowColor: brandColors.magenta,
+  },
+};
```
is this intentionally reversed from the current implementation? on prod we have orange shadow for light and purple shadow for dark

rogervera replied:
Ah yes, good catch. Thank you.  This should be reveresed.
https://gamut.codecademy.com/storybook/?path=/docs/atoms-tooltip--tool-tip as an example.

aresnik11 commented on packages/styleguide/stories/Brand/Atoms/Quote.stories.mdx:
```diff
@@ -17,8 +15,15 @@ Often with an \`Avatar\` or \`Testimonial\`.`,
 
-<Canvas withSource="open">
-  <Story name="Quote">{(args) => <Quote {...args} />}</Story>
+<Canvas withSource="open" className="dark">
```
suggest adding another example here set to light so that it is easier to see what both variations looks like. currently if you toggle, the background doesn't change but the font colors do

rogervera replied:
Got it, so will add a dark wrapper className example to these.

aresnik11 commented on packages/gamut-labs/src/brand/Quote/index.tsx:
```diff
@@ -42,9 +36,13 @@ const Text = styled.q`
   <Container position="relative" mode={mode}>
-    <QuoteIcon alt="" src={mode === 'dark' ? purpleQuotes : orangeQuotes} />
-    <Text>{text}</Text>
+    <QuoteIcon alt="" src={mode === 'light' ? purpleQuotes : orangeQuotes} />
```
is this intentionally reversed so purple quote is for light and orange for dark?

rogervera replied:
Will swap: https://github.com/Codecademy/client-modules/pull/1498#discussion_r597996841

aresnik11 commented on packages/gamut-labs/src/brand/Testimonial/index.tsx:
```diff
@@ -1,75 +1,198 @@
+              width="fit-content"
+              paddingLeft={16}
+            >
+              <Byline
+                firstName={firstName}
+                occupation={occupation}
+                company={company}
+                lastName={lastName}
```
should this be centered?
old testimonial:
<img width="963" alt="Screen Shot 2021-03-19 at 6 52 52 PM" src="https://user-images.githubusercontent.com/8761638/111849591-567ebe80-88e4-11eb-9b12-46638eb63471.png">

new testimonial:
<img width="970" alt="Screen Shot 2021-03-19 at 6 52 57 PM" src="https://user-images.githubusercontent.com/8761638/111849594-57afeb80-88e4-11eb-9cbe-af03df5b1b25.png">

rogervera replied:
Will look into having the Byline spans their content width.

aresnik11 commented on packages/gamut-labs/src/brand/Testimonial/index.tsx:
```diff
@@ -1,75 +1,198 @@
+    <TestimonialWrapper position="relative" mode={mode}>
+      <TestimonialCard>
+        <FlexBox
+          paddingX={24}
+          paddingTop={32}
+          paddingBottom={48}
```
the old testimonial has more padding at the larger viewports, should we do that here as well?

rogervera replied:
Will add for the respective viewport!

aresnik11 reviewed (commented):
I'm super excited about this! It's coming along nicely but I have a bunch of questions/comments/nits that I scattered throughout, mostly around slight differences between this and the existing testimonial that aren't necessarily wrong but I just wanted to flag. Some more here:

1. The original testimonial has an animation on hover that isn't included here, was that intentional? https://github.com/Codecademy/client-modules/blob/main/packages/gamut-labs/src/brand/Testimonial/styles.module.scss#L79-L82. also that entire testimonial styles file can probably be deleted now 😬 
2. The original small testimonial has a different layout than this one, is that intentional? this one has the avatar/byline side by side with the quote vs the old one had the avatar/byline on top of the quote
old testimonial:
<img width="405" alt="Screen Shot 2021-03-19 at 7 17 34 PM" src="https://user-images.githubusercontent.com/8761638/111850852-c6db0f00-88e7-11eb-856b-a4f1cb66086e.png">
new testimonial:
<img width="477" alt="Screen Shot 2021-03-19 at 7 17 29 PM" src="https://user-images.githubusercontent.com/8761638/111850858-cc385980-88e7-11eb-99c2-ab29b48f49a7.png">

rogervera commented:
> I'm super excited about this! It's coming along nicely but I have a bunch of questions/comments/nits that I scattered throughout, mostly around slight differences between this and the existing testimonial that aren't necessarily wrong but I just wanted to flag. Some more here:
> 
> 1. The original testimonial has an animation on hover that isn't included here, was that intentional? https://github.com/Codecademy/client-modules/blob/main/packages/gamut-labs/src/brand/Testimonial/styles.module.scss#L79-L82. also that entire testimonial styles file can probably be deleted now 😬
> 2. The original small testimonial has a different layout than this one, is that intentional? this one has the avatar/byline side by side with the quote vs the old one had the avatar/byline on top of the quote
>    old testimonial:
> 
> <img alt="Screen Shot 2021-03-19 at 7 17 34 PM" width="405" src="https://user-images.githubusercontent.com/8761638/111850852-c6db0f00-88e7-11eb-856b-a4f1cb66086e.png">
> 
> new testimonial:
> <img alt="Screen Shot 2021-03-19 at 7 17 29 PM" width="477" src="https://user-images.githubusercontent.com/8761638/111850858-cc385980-88e7-11eb-99c2-ab29b48f49a7.png">

1. On hover, we can add that back. 

> 2\. The original small testimonial has a different layout than this one, is that intentional? this one has the avatar/byline side by side with the quote vs the old one had the avatar/byline on top of the quote
>     old testimonial:


2. @aresnik11 and I paired on this. The `min-width` on the Quote Container needs to be set to something else than `min-content`. And I'll put up PRs for the migration for comparisons on the monolith and static sites usage spots for the testimonial.

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.35-alpha.93a56e.0<br/>@codecademy/gamut-labs@13.1.5-alpha.93a56e.0<br/>@codecademy/styleguide@33.3.3-alpha.93a56e.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://605c88ca7204d900908b4c17--gamut-preview.netlify.app>https://605c88ca7204d900908b4c17--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/605c88ca7204d900908b4c17>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.1-alpha.aa0a1d.0<br/>@codecademy/gamut-labs@13.3.1-alpha.aa0a1d.0<br/>@codecademy/styleguide@33.7.2-alpha.aa0a1d.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606e07ce80a3bf2643c4bd9b--gamut-preview.netlify.app>https://606e07ce80a3bf2643c4bd9b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606e07ce80a3bf2643c4bd9b>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.1-alpha.582263.0<br/>@codecademy/gamut-labs@13.3.1-alpha.582263.0<br/>@codecademy/styleguide@33.7.2-alpha.582263.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606e1d5d4983973c9704ce2b--gamut-preview.netlify.app>https://606e1d5d4983973c9704ce2b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606e1d5d4983973c9704ce2b>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.1-alpha.3d003a.0<br/>@codecademy/gamut-labs@13.3.1-alpha.3d003a.0<br/>@codecademy/styleguide@33.7.2-alpha.3d003a.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606f1814efa402f248d1ebc0--gamut-preview.netlify.app>https://606f1814efa402f248d1ebc0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606f1814efa402f248d1ebc0>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.6-alpha.daef87.0<br/>@codecademy/gamut-labs@13.3.6-alpha.daef87.0<br/>@codecademy/styleguide@33.8.2-alpha.daef87.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://606f267d7cbf9401507755c0--gamut-preview.netlify.app>https://606f267d7cbf9401507755c0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/606f267d7cbf9401507755c0>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.6-alpha.84507b.0<br/>@codecademy/gamut-labs@13.3.6-alpha.84507b.0<br/>@codecademy/styleguide@33.8.2-alpha.84507b.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070b383533b4a53e5b54122--gamut-preview.netlify.app>https://6070b383533b4a53e5b54122--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070b383533b4a53e5b54122>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.6-alpha.4ff6ee.0<br/>@codecademy/gamut-labs@13.3.6-alpha.4ff6ee.0<br/>@codecademy/styleguide@33.8.2-alpha.4ff6ee.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070b9a7093f7e51d5709c07--gamut-preview.netlify.app>https://6070b9a7093f7e51d5709c07--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070b9a7093f7e51d5709c07>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.6-alpha.e295ba.0<br/>@codecademy/gamut-labs@13.3.6-alpha.e295ba.0<br/>@codecademy/styleguide@33.8.2-alpha.e295ba.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070c14ddd12ea56c63fbd92--gamut-preview.netlify.app>https://6070c14ddd12ea56c63fbd92--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070c14ddd12ea56c63fbd92>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.8-alpha.13400f.0<br/>@codecademy/gamut-labs@13.3.8-alpha.13400f.0<br/>@codecademy/styleguide@33.8.5-alpha.13400f.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070c418a3287262c56138f6--gamut-preview.netlify.app>https://6070c418a3287262c56138f6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070c418a3287262c56138f6>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.8-alpha.8fc3c3.0<br/>@codecademy/gamut-labs@13.3.8-alpha.8fc3c3.0<br/>@codecademy/styleguide@33.8.5-alpha.8fc3c3.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070c872ba6b5a00bf1e64aa--gamut-preview.netlify.app>https://6070c872ba6b5a00bf1e64aa--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070c872ba6b5a00bf1e64aa>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.9-alpha.35a043.0<br/>@codecademy/gamut-labs@13.4.1-alpha.35a043.0<br/>@codecademy/styleguide@33.9.1-alpha.35a043.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070cb0a872c9400b81759cf--gamut-preview.netlify.app>https://6070cb0a872c9400b81759cf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070cb0a872c9400b81759cf>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.9-alpha.187534.0<br/>@codecademy/gamut-labs@13.4.1-alpha.187534.0<br/>@codecademy/styleguide@33.9.1-alpha.187534.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070dc68fe86ed164ce9ca18--gamut-preview.netlify.app>https://6070dc68fe86ed164ce9ca18--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070dc68fe86ed164ce9ca18>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.9-alpha.3d40a2.0<br/>@codecademy/gamut-labs@13.4.1-alpha.3d40a2.0<br/>@codecademy/styleguide@33.9.1-alpha.3d40a2.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.10-alpha.aafefc.0<br/>@codecademy/gamut-labs@13.4.2-alpha.aafefc.0<br/>@codecademy/styleguide@33.9.2-alpha.aafefc.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607376231217934e6b58b024--gamut-preview.netlify.app>https://607376231217934e6b58b024--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607376231217934e6b58b024>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

aresnik11 commented on packages/gamut-labs/src/brand/Byline/index.tsx:
```diff
@@ -10,19 +11,25 @@ export type BylineProps = {
   lastName?: string;
 };
 
+const StyledFlexBox = styled(FlexBox)`
+  width: min-content;
+`;
```
is this not an option for the `width` prop within `FlexBox`?

aresnik11 reviewed (commented):
This is looking so good!!! Not going to officially approve since I know Timea had some design feedback that still needs to be incorporated, but once that is in I'll take another look!

christian-dinh commented:
is this still pending design feedback?

rogervera commented:
> is this still pending design feedback?

Currently yes. Pending changes are for addressing these points from Tímea! Will tag you and @aresnik11 for the check when in. :)
![testimonial notes](https://user-images.githubusercontent.com/17210163/115601461-c0202d00-a2d5-11eb-8847-8b4c3e12bae8.png)

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.48-alpha.b90b1d.0<br/>@codecademy/gamut-labs@14.3.2-alpha.b90b1d.0<br/>@codecademy/styleguide@36.3.2-alpha.b90b1d.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086c699e0d27300a2bae38a--gamut-preview.netlify.app>https://6086c699e0d27300a2bae38a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086c699e0d27300a2bae38a>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.52-alpha.e185d4.0<br/>@codecademy/gamut-labs@14.4.4-alpha.e185d4.0<br/>@codecademy/styleguide@36.4.4-alpha.e185d4.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6088182bfa8e8b483f646d2e--gamut-preview.netlify.app>https://6088182bfa8e8b483f646d2e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6088182bfa8e8b483f646d2e>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.0f785b.0<br/>@codecademy/gamut-labs@16.0.2-alpha.0f785b.0<br/>@codecademy/styleguide@37.5.5-alpha.0f785b.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609c31d2d2553e18e1a0384e--gamut-preview.netlify.app>https://609c31d2d2553e18e1a0384e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609c31d2d2553e18e1a0384e>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.e5fd8b.0<br/>@codecademy/gamut-labs@16.0.2-alpha.e5fd8b.0<br/>@codecademy/styleguide@37.5.5-alpha.e5fd8b.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609c347f8f8eb119f0e4ebe0--gamut-preview.netlify.app>https://609c347f8f8eb119f0e4ebe0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609c347f8f8eb119f0e4ebe0>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.946780.0<br/>@codecademy/gamut-labs@16.0.2-alpha.946780.0<br/>@codecademy/styleguide@37.5.5-alpha.946780.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609c3d1ea957662d125ffa58--gamut-preview.netlify.app>https://609c3d1ea957662d125ffa58--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609c3d1ea957662d125ffa58>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.c672cc.0<br/>@codecademy/gamut-labs@16.0.2-alpha.c672cc.0<br/>@codecademy/styleguide@37.5.5-alpha.c672cc.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609c48c2d2553e3027a03896--gamut-preview.netlify.app>https://609c48c2d2553e3027a03896--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609c48c2d2553e3027a03896>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.ea2cd6.0<br/>@codecademy/gamut-labs@16.0.2-alpha.ea2cd6.0<br/>@codecademy/styleguide@37.5.5-alpha.ea2cd6.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609c5decf061ec00940ea61f--gamut-preview.netlify.app>https://609c5decf061ec00940ea61f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609c5decf061ec00940ea61f>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

JoshuaKGoldberg commented:
TBH I think @codecaaron is better positioned than me to review these - but lmk if I should step back up Aaron!

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.97-alpha.6b246a.0<br/>@codecademy/gamut-labs@16.0.2-alpha.6b246a.0<br/>@codecademy/styleguide@37.5.5-alpha.6b246a.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609d668e4e929e3a68b59515--gamut-preview.netlify.app>https://609d668e4e929e3a68b59515--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609d668e4e929e3a68b59515>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609e7ac8592700df7b45e145--gamut-preview.netlify.app>https://609e7ac8592700df7b45e145--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609e7ac8592700df7b45e145>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.102-alpha.cb9ccf.0<br/>@codecademy/gamut-labs@17.0.1-alpha.cb9ccf.0<br/>@codecademy/styleguide@38.0.1-alpha.cb9ccf.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609e7f5d155fddecfb26769e--gamut-preview.netlify.app>https://609e7f5d155fddecfb26769e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609e7f5d155fddecfb26769e>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

JoshuaKGoldberg commented on packages/gamut-labs/src/Testimonial/index.tsx:
```diff
@@ -1,77 +1,179 @@
+   */
+  longQuote?: string;
+  href?: string;
+  /** Click callback */
```
Nit: you don't say!

```suggestion
```

JoshuaKGoldberg reviewed (approved):
Great to see progress on this 😁  love it!

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.103-alpha.356ac8.0<br/>@codecademy/gamut-labs@17.1.1-alpha.356ac8.0<br/>@codecademy/styleguide@38.1.1-alpha.356ac8.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a279b1be2e5110096f1f5a--gamut-preview.netlify.app>https://60a279b1be2e5110096f1f5a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a279b1be2e5110096f1f5a>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.103-alpha.eea689.0<br/>@codecademy/gamut-labs@17.1.1-alpha.eea689.0<br/>@codecademy/styleguide@38.1.1-alpha.eea689.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a2dd189b1e03345b167b12--gamut-preview.netlify.app>https://60a2dd189b1e03345b167b12--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a2dd189b1e03345b167b12>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

aresnik11 commented on packages/gamut-labs/src/Testimonial/index.tsx:
```diff
@@ -1,77 +1,178 @@
+              </Box>
+            </FlexBox>
+            <Box flexGrow={9999} flexShrink={1} flexBasis="0" minWidth="70%">
+              <Box position="relative">
+                <QuoteIcon
+                  alt=""
+                  width="109px"
+                  height="74px"
+                  src={modes[mode].quoteIcon}
+                />
+                <QuoteText>
+                  {longQuote ? (
+                    <>
+                      <Box as="span" display={{ _: 'initial', sm: 'none' }}>
+                        {quote}
+                      </Box>
+                      <Box as="span" display={{ _: 'none', sm: 'initial' }}>
+                        {longQuote}
+                      </Box>
+                    </>
+                  ) : (
+                    quote
+                  )}
+                </QuoteText>
+              </Box>
```
curious why you decided to inline this instead of using the `Quote` component we previously had

aresnik11 commented on packages/gamut-labs/src/Testimonial/index.tsx:
```diff
@@ -1,77 +1,178 @@
+                />
+              </Box>
+            </FlexBox>
+            <Box flexGrow={9999} flexShrink={1} flexBasis="0" minWidth="70%">
```
i'm not the best at `flex` yet but `flexGrow={9999}` seems icky. @christian-dinh @JoshuaKGoldberg can i get a sanity check here?

christian-dinh replied:
ah yeah this was my suggestion. it's definitely a [bit of a hack](https://joren.co/flex-grow-9999-hack/)

basically what this does (along with the `flexBasis={0}` in the left column's `FlexBox`) is always expand the right column while always shrinking the left column if they are side-by-side, and always expanding both if the right column wraps to a new line.

christian-dinh replied:
might be good to include a comment here if we're ok with doing it this way

aresnik11 replied:
ohhh i see, that link was super helpful

aresnik11 commented on packages/gamut-labs/src/Testimonial/index.tsx:
```diff
@@ -1,77 +1,178 @@
+              <Box position="relative">
+                <QuoteIcon
+                  alt=""
+                  width="109px"
+                  height="74px"
```
are these needed? they weren't included when we were using the `Quote` component

judahtanthony replied:
yes.  They were requested by Timea.

aresnik11 commented on packages/gamut-labs/src/Testimonial/index.tsx:
```diff
@@ -1,77 +1,178 @@
+                  <Avatar alt="" src={imageUrl} mode={mode} />
+                </Box>
+              )}
+              <Box mb={16} mr={16} minWidth="11rem">
```
any reason why this is specifically `11rem`? on iphone x and smaller, the byline falls to below the avatar. if we reduced this to `9.5rem` it would not do this, or even `10rem` would cover more of the phones
<img width="368" alt="Screen Shot 2021-05-18 at 8 59 53 PM" src="https://user-images.githubusercontent.com/8761638/118741329-06f13c00-b81c-11eb-80c6-66487c33c9d9.png">

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.103-alpha.e69b98.0<br/>@codecademy/gamut-labs@17.1.1-alpha.e69b98.0<br/>@codecademy/styleguide@38.1.1-alpha.e69b98.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a54504a1b0d618db82f1c2--gamut-preview.netlify.app>https://60a54504a1b0d618db82f1c2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a54504a1b0d618db82f1c2>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.106-alpha.b047fb.0<br/>@codecademy/gamut-labs@17.3.1-alpha.b047fb.0<br/>@codecademy/styleguide@38.3.1-alpha.b047fb.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a54c6068578f1f62bbb5a5--gamut-preview.netlify.app>https://60a54c6068578f1f62bbb5a5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a54c6068578f1f62bbb5a5>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.107-alpha.319059.0<br/>@codecademy/gamut-labs@17.3.2-alpha.319059.0<br/>@codecademy/styleguide@38.3.2-alpha.319059.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a55ffa27170d37aace3255--gamut-preview.netlify.app>https://60a55ffa27170d37aace3255--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a55ffa27170d37aace3255>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->
